Use multi Tx(s) for Branch and Tag Push

harry-hov opened the pull request:
...

harry-hov committed:
Use multi Tx(s) for Branch and Tag Push

Broadcasting repeated transactions for multiple branches/tags push
can cause incorrect sequence number error. The one solution for
that is to use BROADCAST_MODE_BLOCK. Which leads to slower response.

Another problem is unwanted large gas consumptions. To mitigate
that, use single tx to push multiple branches/tags.

Signed-off-by: Hariom Verma <hariom@gitopia.org>

harry-hov committed:
Get rid of delete remote branch warning

When the push is complete, outputs one or more ok <dst>
or error <dst> <why>? lines to indicate success or failure of each
pushed ref.

Current push outputs <src> instead of <dst>, which leads to
"warning: helper reported unexpected status of..."

Signed-off-by: Hariom Verma <hariom@gitopia.org>

faza commented on cmd/git-remote-gitopia/gitopia.go:
```diff
@@ -243,73 +221,135 @@ func (h *GitopiaHandler) Push(remote *core.Remote, local string, remoteRef strin
+		if err != nil {
+			return nil, err
+		}
 	}
```
Sign and broadcast a single transaction with multiple messages

harry-hov committed:
Sign and broadcast with multiple messages

Multiple Txs can happen simultaneously. e.g someone decides to push
branch and tag together, in that two transaction happens.
Currently, We are calling signAndBroadcastTx() for every msg. Since
we can broadcast multiple msg in a single Tx. Lets call
signAndBroadcastTx() only once.

Signed-off-by: Hariom Verma <hariom@gitopia.org>

harry-hov committed:
Bump gitopia version

Changes:
  - HaveBranchPermission() is replaced by generalized HavePermission()

Signed-off-by: Hariom Verma <hariom@gitopia.org>
